roles/migrationcontroller/tasks/main.yml

```diff
@@ -56,26 +56,30 @@
       namespace: kube-system
     register: cluster_configmap
```

Comment:
Yes, I think it makes sense to use infrastructure or the clusterversion resource you also pointed out. Since the clusterversion stores the exact OCP version it might make better sense to look for that.

We also need to test all of this on OCP 3. I'm not sure if using k8s facts to get non-existent resource types will cause errors. We know how to check for the api resources though, so we can adjust if needed.

https://github.com/ansibleplaybookbundle/mediawiki-apb/blob/master/tasks/main.yml#L24 would be an example of that.

Reply:
@jmontleon looks like the module does not error out on missing kube API I tried a sample run with the following


```
[alpatel@alpatel workloads]$ oc project
Using project "openshift-migration" on server "https://master.alpatel-image-3110.mg.dog8code.com:443".
[alpatel@alpatel workloads]$ ansible-playbook /tmp/play.yml
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'

PLAY [localhost] ************************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************************
ok: [localhost]

TASK [Get the infrastructure] ***********************************************************************************************************************************************************
ok: [localhost]

TASK [debug] ****************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": {
        "changed": false,
        "failed": false,
        "resources": []
    }
}

PLAY RECAP ******************************************************************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0

[alpatel@alpatel workloads]$ cat /tmp/play.yml
---
- hosts: localhost
  tasks:
  - name: Get the infrastructure
    k8s_info:
      api_version: config.openshift.io/v1
      kind: Infrastructure
      name: cluster
    register: infrastructures
    # when: "{{ True if 'config.openshift.io' in lookup('k8s', cluster_info='api_groups') else False }}"

  - debug: msg="{{ infrastructures }}"
[alpatel@alpatel workloads]$
```